kubemarine/plugins/__init__.py

```diff
@@ -1018,14 +1029,14 @@ def apply_source(cluster: KubernetesCluster, config: dict) -> None:
     else:
         apply_common_group = cluster.create_group_from_groups_nodes_names(apply_groups, apply_nodes)
 
-    destination_common_group.put(source, destination_path, backup=True, sudo=use_sudo)
+    destination_common_group.put(source, destination_path, backup=True, sudo=use_sudo, dry_run=dry_run)
 
     if apply_required:
         method = apply_common_group.run
         if use_sudo:
             method = apply_common_group.sudo
         cluster.log.debug("Applying yaml...")
```

Comment:
Minor moment, rendering of plugins' templates might depend on previous shell commands. See `runtime_vars`. If it is possible, I suggest to skip rendering of templates in such cases, because its result will likely be not the same as during the real run.